application-confluence-migrator-pro-ui/src/main/resources/ConfluenceMigratorPro/Code/CommonCode.xml

```diff
@@ -178,6 +187,36 @@
       $('a[href$="#' + $(this).attr('id') + '"]').find('.cfmTitleIcon').removeClass('active');
     }
   });
+
+  $(function() {
+    // Since the associated liveData is refreshed on file upload, we don't need a response container.
+    new XWiki.FileUploader($('#AddAttachment input[type=file]')[0], {
```

Comment:
Isn't this selector too generic? How do you know it's the upload for Confluence packages and not the one from the Attachments document extra tab or something in a panel?

Reply:
It wasn't actually as easy as I would have expected
Initially I left the generic `AddAttachment` id since I though that maybe, when it was added, it was known that the attachments tab won't be displayed (the same id is used in there). Now, because of this new id, I noticed some things
It looks like whenever you are uploading something using `upload.js`, this will trigger the add `attachments.min.js` file (couldn't find why though, I don't see it in the code) and [this](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/viewers/attachments.js#L234) listener will be triggered. Luckily, this is generic so it's not needed to handle anymore the delete myself. But I had to add the `attachments.js` code always.
Unfortunately, [this code](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/viewers/attachments.js#L280) was not made generic enough, and it actually assumes that `#docAttachments` is on the page. This will produce some errors on the browser console, which aren't affecting the functionality. I will fix this in the platform, but the error will remain for now in older xwiki versions.

Reply:
I found the reason. It's because when in `upload.js` the post request is [send](https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-14.10.2/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/widgets/upload.js#L219), it will receive the `ConfluenceMigratorPro/Code/PackagesStore?viewer=attachments` response, and this is the one adding the `attachments.js`

For `the attachments.js`,  when I added https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-14.10.2/xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/attachment_macros.vm#L250 it was meant and tested to be used also outside of the Attachments tab, or multiple times on a page, as a normal macro. But it wasn't indeed made to be used along with the upload feature.
Now, the attachments tab is using a different `XWiki.FileUpload` instance, so the only problem would be with the events, as it had happened (since I did not checked who produced the upload event..). I think it should be safe to use it